src/models/transaction.rs

```diff
@@ -1,68 +1,75 @@
-                to: rpc_transaction.to.map_or_else(|| TransactionKind::Create, TransactionKind::Call),
-                value: rpc_transaction.value,
-                input: rpc_transaction.input,
-                chain_id: rpc_transaction.chain_id.map(|id| id.to::<u64>()),
-            })),
-            1 => Ok(reth_primitives::Transaction::Eip2930(TxEip2930 {
-                chain_id: rpc_transaction.chain_id.ok_or(ConversionError)?.to::<u64>(),
-                nonce: rpc_transaction.nonce.to::<u64>(),
-                gas_price: rpc_transaction.gas_price.ok_or(ConversionError)?.to::<u128>(),
-                gas_limit: rpc_transaction.gas.try_into().map_err(|_| ConversionError)?,
-                to: rpc_transaction.to.map_or_else(|| TransactionKind::Create, TransactionKind::Call),
-                value: rpc_transaction.value,
-                access_list: AccessList(
-                    rpc_transaction
-                        .access_list
-                        .unwrap_or_default()
-                        .into_iter()
-                        .map(|access_list| AccessListItem {
-                            address: access_list.address,
-                            storage_keys: access_list.storage_keys,
-                        })
-                        .collect(),
-                ),
-                input: rpc_transaction.input,
-            })),
-            2 => Ok(reth_primitives::Transaction::Eip1559(TxEip1559 {
-                chain_id: rpc_transaction.chain_id.ok_or(ConversionError)?.to::<u64>(),
-                nonce: rpc_transaction.nonce.to::<u64>(),
-                gas_limit: rpc_transaction.gas.try_into().map_err(|_| ConversionError)?,
-                max_fee_per_gas: rpc_transaction.max_fee_per_gas.ok_or(ConversionError)?.to::<u128>(),
-                max_priority_fee_per_gas: rpc_transaction.max_priority_fee_per_gas.ok_or(ConversionError)?.to::<u128>(),
-                to: rpc_transaction.to.map_or_else(|| TransactionKind::Create, TransactionKind::Call),
-                value: rpc_transaction.value,
-                access_list: AccessList(
-                    rpc_transaction
-                        .access_list
-                        .unwrap_or_default()
-                        .into_iter()
-                        .map(|access_list| AccessListItem {
-                            address: access_list.address,
-                            storage_keys: access_list.storage_keys,
-                        })
-                        .collect(),
-                ),
-                input: rpc_transaction.input,
-            })),
-            _ => Err(ConversionError.into()),
-        },
+    let tx_type = match rpc_transaction.transaction_type {
+        Some(transaction_type) => {
+            let transaction_type_u8: u8 = transaction_type.try_into().map_err(|_| ConversionError)?;
+
+            transaction_type_u8.try_into().map_err(|_| ConversionError)?
```

Comment:
nit: I don't know if in Rust you would add `_u8`

Reply:
you are right, just changed it